docs/02.API-REFERENCE.md

```diff
@@ -261,6 +261,78 @@ jerry_init (jerry_init_flag_t flags)
 - [jerry_cleanup](#jerry_cleanup)
 
 
+## jerry_init_with_user_context
+
+**Summary**
+
+Initializes the JerryScript engine, making it possible to run JavaScript code and perform operations
+on JavaScript values. In addition to the first parameter, this function accepts two more parameters
+with which it is possible to store a `void *` pointer inside the context being initialized. The
+function calls the callback given in its `init_cb` parameter to allocate the memory for the pointer
+and it stores the function pointer given in the `deinit_cb` parameter along with the pointer so that
+it may be called to free the stored pointer when the context is discarded.
+
```

Comment:
Could you please describe the relationship among `jerry_init_with_user_context`, `jerry_init` and `jerry_cleanup`.
e.g.
in the `##jerry_init`: Note: If a user context need to be initialized, please use `jerry_init_with_user_context` instead.
in the `##jerry_init_with_user_context`: Note: `jerry_init` will be called inside `jerry_init_with_user_context`, and `deinit_cb` will be called in the `jerry_cleanup`
in the `##jerry_cleanup`: If the engine is Initialized by `jerry_init_with_user_context` with `deinit_cb`, then the `deinit_cb` will be called during cleanup.

What do you think?